extensions/apps/extensions/src/components/pages/my-extensions/extension-element.tsx

```diff
@@ -0,0 +1,251 @@
+    switch (applicationType) {
+      case AkashaAppApplicationType.App:
+        return <App />;
+      case AkashaAppApplicationType.Plugin:
+        return <Plugin />;
+      case AkashaAppApplicationType.Widget:
+        return <Widget />;
+    }
+  };
+
+  const getExtensionStatus = () => {
+    if (appStreamData?.edges?.length === 0) {
+      return ExtensionStatus.Draft;
+    }
+    if (appStreamData?.edges[0]) {
+      if (appStreamData?.edges[0].node?.status === null) {
+        return ExtensionStatus.Draft;
+      }
+      if (appStreamData?.edges[0].node?.status === AkashaAppsStreamModerationStatus.InReview) {
+        return ExtensionStatus.Pending;
+      }
+      if (appStreamData?.edges[0].node?.status === AkashaAppsStreamModerationStatus.Ok) {
+        return ExtensionStatus.Published;
+      }
+    }
+  };
+
+  const getStatusIndicatorStyle = () => {
+    switch (getExtensionStatus()) {
+      case ExtensionStatus.Draft:
+        return 'bg-grey6';
+      case ExtensionStatus.Published:
+        return 'bg-success';
+      case ExtensionStatus.Pending:
+        return 'bg-(errorLight dark:errorDark)';
+      default:
+        return 'bg-grey6';
+    }
+  };
+
+  const handleExtensionRemove = () => {
+    navigateToModal({
+      name: `remove-extension-confirmation`,
+      extensionId: extensionData.id,
+    });
+  };
+
+  const menuItems = (extensionStatus: string) => {
+    switch (extensionStatus) {
+      case ExtensionStatus.Pending:
+        return [
+          {
+            label: t('Check status'),
+            icon: <ClockIcon />,
+            onClick: () => {},
+          },
+        ] as MenuProps['items'];
+      case ExtensionStatus.Published:
+        return [
+          {
+            label: t('View'),
+            icon: <EyeIcon />,
+            onClick: () => {},
+          },
+          {
+            label: t('Edit'),
+            icon: <PencilIcon />,
+            onClick: () => {},
+          },
+          {
+            label: t('Unpublish'),
+            icon: <XMarkIcon />,
+            onClick: () => {},
+            color: { light: 'errorLight', dark: 'errorDark' },
+          },
+        ] as MenuProps['items'];
+      case ExtensionStatus.Draft:
+        return [
+          {
+            label: t('Publish'),
+            icon: <ArrowUpOnSquareIcon />,
+            onClick: () => {},
+          },
+          {
+            label: t('Edit'),
+            icon: <PencilIcon />,
+            onClick: () => {},
+          },
+          {
+            label: t('Delete'),
+            icon: <XMarkIcon />,
+            onClick: handleExtensionRemove,
+            color: { light: 'errorLight', dark: 'errorDark' },
+          },
+        ] as MenuProps['items'];
+      default:
+        return [];
+    }
+  };
+
+  const showElement = () => {
+    if (filter.id === '0') {
+      return true;
+    }
+    return filter.title === getExtensionStatus();
+  };
+
+  return (
+    <>
+      {showElement() && (
```

Comment:
```suggestion
      {filter.id === '0' || filter.title === getExtensionStatus() && (
```

Reply:
I think the function is more explicit in this case